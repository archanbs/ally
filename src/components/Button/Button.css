.global-button {
    padding: 6px;
    cursor: pointer;
    border-radius: 5px;
}