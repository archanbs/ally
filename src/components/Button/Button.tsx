import React from "react";
import "./Button.css";

export interface IButtonProps {
  id?: string;
  type?: "button" | "submit" | "reset";
  onClick?: (ev) => void;
  text: string;
  className?: string;
  children?: React.ReactNode;
}

export const Button = function ({ id, text, onClick, type, className, children }: IButtonProps) {
  const onButtonClick = (ev) => {
    console.log("Button clicked");
    ev.preventDefault();
    onClick?.(ev);
  };

  return (
    <button id={id} className={`global-button${className !== undefined ? ` ${className}` : ""}`} type={type} onClick={onButtonClick}>
      {text}
      {children}
    </button>
  );
};
