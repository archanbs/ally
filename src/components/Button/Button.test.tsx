import renderer from "react-test-renderer";
import { Button } from "./Button";
import React from "react";

test("Button snapshot test", () => {
  const compTree = renderer.create(<Button type="button" id="test" text="Button" className="font-bold" />).toJSON();
  expect(compTree).toMatchSnapshot();
});

test("Button snapshot test with children", () => {
  const compTree = renderer
    .create(<Button type="button" id="test" text="Button" className="text-[10px]" children={<span>Test child</span>} />)
    .toJSON();
  expect(compTree).toMatchSnapshot();
});
