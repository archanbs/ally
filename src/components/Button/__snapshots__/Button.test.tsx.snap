// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot test 1`] = `
<button
  className="global-button font-bold"
  id="test"
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button snapshot test with children 1`] = `
<button
  className="global-button font-bold"
  id="test"
  onClick={[Function]}
  type="button"
>
  Button
  <span>
    Test child
  </span>
</button>
`;
